src/metabase/models/dashboard.clj

```diff
@@ -180,7 +180,7 @@
 (defmethod revision/serialize-instance :model/Dashboard
   [_model _id dashboard]
   (-> dashboard
-      (select-keys [:description :name :cache_ttl :auto_apply_filters])
+      (select-keys [:collection_id :description :name :cache_ttl :auto_apply_filters])
```

Comment:
we probably forgot to do this. We do track collection changes for Card, so we should track for Dashboard too.